shared/fs/container.js

```diff
@@ -12,28 +12,46 @@ type OwnProps = {
 
 type StateProps = {
   path: Types.Path,
-  items: I.List<Types.Path>,
+  items: I.List<Types.PathItem>,
   progress: Types.ProgressType,
+  sortSetting: Types.SortSetting,
 }
 
-type DispatchProps = {}
+type DispatchProps = {
+  _sortSettingChangeFactory: (path: Types.Path) => (setting: Types._SortSetting) => void,
+}
 
 const mapStateToProps = (state: TypedState, {routeProps}: OwnProps) => {
   const path = Types.stringToPath(routeProps.get('path', Constants.defaultPath))
+  const sortSetting = state.fs.pathUserSettings.get(path, Constants.makePathUserSetting()).get('sort')
```

Comment:
don't do thinks like sorting in mapStateToProps. any derived stuff should be in mergeProps

Reply:
This particular line just gets the `sortSetting`. The actually does happen in `mergeProps` below.